### DISCUSSION TITLE
Shared FS snaphot repository support

### DISCUSSION THREAD

**Discussion by iNode** [Opster/opensearch-k8s-operator#390] [Q&A] [answered]
Hi Team,

I'm looking into OpenSearch cluster deployment using operator and not sure that operator configuration covers my use case. Is it possible to configure shared snapshot repository (for example NFS one) on the nodes?

It'd be good to have possibility to specify additional/custom volumeMount for this to use it with FS repository type. We don't want to share default one which maps `data` storage to the `/usr/share/opensearch/data`.

As per CRD specs/documentation it looks like only ConfigMaps and Secrets can be mounted additionally to `persistence` configuration at the moment.

If this capability is not present at the moment then is it something that can be implemented in the operator?

**Comment by swoehrl-mw** [comment, marked as answer]
Hi @iNode. Great to hear you are considering the operator.

Currently a feature is in the works (PR #370, Issue #278) to allow configuration of snapshot repositories via the operator. 

As you've already gleaned from the docs additional volumes can be mounted to the pods, but they are currently limited to secrets and configmaps. There is an open feature request (#275) to support CSI for additional volumes. Would that (with NFS via CSI) cover your usecase?

**Reply by iNode** [reply]
HI @swoehrl-mw. Thanks for a quick response. It looks relevant and promising. 

I think that this feature (#275) should cover mountVolume part of use case, and PR (#370) should simplify configuration part of the story if implementation will take FS type of repository in account as well.

Will keep an eye on them and see if we can contribute something. Thanks!
